src/components/picker/helpers/usePickerMode.ts

```diff
@@ -0,0 +1,47 @@
+import {ExpandableOverlayProps} from 'src/incubator';
+import {PickerProps, PickerModeTypes} from '../index';
+import {CustomPickerProps} from '../types';
+
+type PickerType = {
```

Comment:
Why declaring the type again and not using PickerModeTypes?

Reply:
`PickerModeTypes` is an enum, if I want to use the enum I need to add interface to convert the keys/values into type that holds boolean value.
Am I wrong ?

Reply:
I think the booleans are redundant. You're only using them in the else statement to extract type from deprecated props. You can simply do this for example (better in a dedicated function): 
`if (useWheelPicker) type = PickerModeTypes.WheelPicker`

Reply:
as we spoke I'll leave it for now

Reply:
For the sack of clearity I wounld change the boolead type to be `isModal`|`isDialog` etc. and in Picker component the check will be `type.isDialog` etc. It will be cleared.
`PickerModeBooleans` I would change to `PickerTypesBooleans`